include/rocksdb/flush_manager.h

```diff
@@ -0,0 +1,36 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <functional>
+#include <vector>
+
+namespace rocksdb {
+
+class ExternalFlushManager {
```

Comment:
Also we should not call things "External" in public facing API. This naming is friendly to us but not user friendly.